railties/lib/rails/commands/task_builder/test.rb

```diff
@@ -0,0 +1,33 @@
+require_relative 'common'
+require 'rake'
+
+module Rails
+  module Commands
+    module TaskBuilder
+      # This is a wrapper around all Rails test tasks, including:
+      #   rails test
+      #   rails test:db
+      class Test
```

Comment:
Without implying any comment on the overall structure, which I don't think I've fully understood yet... I'd suggest "Core" for that class name.